src/main/kotlin/io/github/rybalkinsd/kohttp/ext/HttpContextExt.kt

```diff
@@ -14,10 +14,21 @@ fun HttpContext.url(url: URL) {
 
     host = url.host ?: throw IllegalArgumentException("unexpected host: $host")
 
-    if (url.port != -1) {
+    if (url.port != -1)
         port = url.port
-    }
+
     path = url.path
+
+    if (url.query?.isNotBlank() == true) {
+        param {
+            url.query.split("&")
+                .map { it.split("=", limit = 2) }
+                .groupBy({ it[0] }, { it.getOrNull(1) })
```

Comment:
probably, grouping op here is not necessary, b/c `ParamContext` applies add in a special way